### ISSUE TITLE
Don't understand logic for segmentation.

### ISSUE THREAD

**Issue by comdiv** [thelastpickle/cassandra-reaper#955]
We have 10-nodes cluster with 3-replica keyspace, each node owns 256 tokens

In keyspace we have data with token, for example `-4964191742879005665`

But when we start full repair for this keyspace reaper shows 290 segments but whose token ranges not covers `-4964191742879005665` at all.
It hase one segment `-51... .. -50...` and `-48..  .. -47` and not any segment includes  `-4964191742879005665`

What's wrong? May be i missunderstand what segment means and how repair works?

**Comment by adejanovski** [comment]
Segments will contain several token ranges starting with Cassandra 2.2, which share the exact same replicas and can then be repaired by Cassandra in a single repair session.
In the UI, we're just showing the "base range", which we needed to keep for compatibility issues, but in the repair_run table theres a column that stores all the token ranges the segment contains.

TL;DR: segments store several token ranges that will be repaired in a single session and all the tokens will be repaired by repair runs.

**Comment by comdiv** [comment]
Not very comfortable, but I understand that all tokens are really processed
